torchdata/datapipes/iter/util/header.py

```diff
@@ -53,3 +56,65 @@ def __len__(self) -> int:
                 "The actual value may be smaller if the actual length of source_datapipe is smaller than the limit."
             )
             return self.limit
+
+
+@functional_datapipe("limit")
+class LimiterIterDataPipe(IterDataPipe[T_co]):
+    """
+    Limits the number of samples that the source DataPipe can yield based the input as well as the number of
+    processes and workers if distribute- or multi-processing is enabled (functional name: ``limit``).
+
+    The number of samples that will be yielded equals ``num_take // (num_processes * worker_info.num_workers)``.
+
+    Args:
+        source_datapipe: IterDataPipe with samples of data that will be yielded
+        num_take: the number of samples that you wish to yield across all processes and workers, should be less than
+            or equal to the ``len(source_datapipe)``
+
+    Example:
+        >>> source_dp = IterableWrapper(range(20))
+        >>> limiter_dp = source_dp.limit(5)
+        >>> list(limiter_dp)
+        [0, 1, 2, 3, 4]
+    """
+
+    def __init__(self, source_datapipe: IterDataPipe[T_co], num_take: int) -> None:
+        super().__init__()
+        self.source_datapipe: IterDataPipe[T_co] = source_datapipe
+        self.num_take: int = num_take
+        self.num_processes: int = 1
+        if dist.is_available() and dist.is_initialized():
+            self.num_processes = dist.get_world_size()
+
+    def _get_num_workers(self):
+        num_workers = self.num_processes
+        worker_info = get_worker_info()
+        if worker_info is not None:
+            # Note that we are assuming each distributed process has the same number of workers
+            num_workers *= worker_info.num_workers
+        return num_workers
+
+    def __iter__(self) -> Iterator:
+        # TODO: pmeier - this is weird as it drops more elements than it should
```

Comment:
@pmeier Can you elaborate on this?

Reply:
I can't since I've not written nor ever used it. cc @fmassa for context.

Reply:
The reason that it drops more elements than it should is because we just need to make sure that every DDP process has the same number of elements, and not every individual worker thread in the dataloader.

This is what `DistributedSampler` does, as the DataLoader properly takes care of `worker_threads` with different number of elements.